packages/frontend/src/components/EditorLayout/index.tsx

```diff
@@ -143,32 +145,32 @@ export default function EditorLayout() {
             </Box>
 
             {!loading && (
-              <EditableTypography
-                variant="body1"
-                onConfirm={onFlowNameUpdate}
-                noWrap
-                sx={{ display: 'flex', flex: 1, maxWidth: '30vw', ml: 1 }}
-              >
-                {flow?.name}
-              </EditableTypography>
+              <EditableInput
+                value={flow?.name}
+                onSave={onFlowNameUpdate}
+                readOnly={isEditorReadOnly}
+                width={isMobile ? '40%' : undefined}
+              />
             )}
           </Flex>
 
-          <TouchableTooltip label="Guide" aria-label="guide tooltip">
-            <IconButton
-              as={Link}
-              to={URLS.GUIDE_LINK}
-              target="_blank"
-              variant="clear"
-              aria-label="guide"
-              icon={<BiInfoCircle />}
-              colorScheme="secondary"
-              _hover={{
-                color: 'primary.500',
-                bg: 'interaction.muted.main.hover',
-              }}
-            />
-          </TouchableTooltip>
+          {!isMobile && (
+            <TouchableTooltip label="Guide" aria-label="guide tooltip">
+              <IconButton
+                as={Link}
+                to={URLS.GUIDE_LINK}
+                target="_blank"
+                variant="clear"
+                aria-label="guide"
+                icon={<BiInfoCircle />}
+                colorScheme="secondary"
+                _hover={{
+                  color: 'primary.500',
+                  bg: 'interaction.muted.main.hover',
+                }}
+              />
+            </TouchableTooltip>
+          )}
```

Comment:
can wrap everything on the right side with a flex and add a gap to it so that the pipe name can only be renamed after hovering over the text thankss